client/scripts/views/components/new_thread_form.ts

```diff
@@ -0,0 +1,386 @@
+import 'components/new_thread_form.scss';
+
+import m, { VnodeDOM } from 'mithril';
+import _ from 'lodash';
+import $ from 'jquery';
+import Quill from 'quill-2.0-dev/quill';
+import { Form, FormGroup, Input, Button, ButtonGroup, Icons, Grid, Col, Tooltip, List, ListItem } from 'construct-ui';
+
+import app from 'state';
+import QuillEditor from 'views/components/quill_editor';
+import AutoCompleteTagForm from 'views/components/autocomplete_tag_form';
+import { detectURL, getLinkTitle, newLink, newThread, saveDraft } from 'views/pages/threads';
+import { OffchainTag } from 'client/scripts/models';
+import QuillFormattedText from './quill_formatted_text';
+import MarkdownFormattedText from './markdown_formatted_text';
+import { confirmationModalWithText } from '../modals/confirm_modal';
+
+interface IState {
+  form: IThreadForm,
+  activeTag: OffchainTag | string,
+  fromDraft?: number,
+  error,
+  quillEditorState,
+  hasComment: boolean,
+  autoTitleOverride,
+  newType: string,
+  uploadsInProgress: number,
+}
+
+interface IThreadForm {
+  tagName?: string;
+  tagId?: number;
+  url?: string;
+  title?: string;
+}
+
+export const populateDraft = async (state, draft) => {
```

Comment:
Would rename to "saveDraft" for consistency....not sure that populate has any particular meaning here that isn't already communicated by using save

Reply:
This actually isn't to save a draft, it's to render/load an already saved draft in the thread form!

Reply:
Maybe "loadDraft"?

Reply:
loadDraft sounds like it makes sense.